velox/functions/prestosql/tests/JsonFunctionsTest.cpp

```diff
@@ -58,6 +60,35 @@ class JsonFunctionsTest : public functions::test::FunctionBaseTest {
   }
 };
 
+TEST_F(JsonFunctionsTest, jsonFormat) {
+  const auto jsonFormat = [&](std::optional<std::string> value) {
+    return evaluateOnce<std::string, std::string>(
+        "json_format(c0)", {value}, {JSON()});
+  };
+
+  EXPECT_EQ(jsonFormat(std::nullopt), std::nullopt);
+  EXPECT_EQ(jsonFormat(R"(true)"), "true");
+  EXPECT_EQ(jsonFormat(R"(null)"), "null");
+  EXPECT_EQ(jsonFormat(R"(42)"), "42");
+  EXPECT_EQ(jsonFormat(R"("abc")"), "\"abc\"");
+  EXPECT_EQ(jsonFormat(R"([1, 2, 3])"), "[1, 2, 3]");
+  EXPECT_EQ(jsonFormat(R"({"k1":"v1"})"), "{\"k1\":\"v1\"}");
+
+  auto data = makeRowVector({makeFlatVector<StringView>(
+      {"This is a long sentence", "This is some other sentence"})});
+
+  auto result = evaluate("json_format(c0)", data);
+  auto expected = makeFlatVector<StringView>(
+      {"This is a long sentence", "This is some other sentence"});
+  facebook::velox::test::assertEqualVectors(expected, result);
+
+  data = makeRowVector({makeConstant("apple", 2)});
+  result = evaluate("json_format(c0)", data);
+  expected = makeFlatVector<StringView>({{"apple", "apple"}});
+
+  facebook::velox::test::assertEqualVectors(expected, result);
```

Comment:
These are nice tests. Let's also add a test with an "if". This test should fail if you revert the code back to when it was not using moveOrCopyResult.